src/map/label.cpp

```diff
@@ -550,11 +555,12 @@ void terrain_label::draw()
 	bool use_markup = color_ == font::LABEL_COLOR;
 
 	font::floating_label flabel(text_.str());
+	flabel.set_font_size(scale_to_map_zoom(font::SIZE_NORMAL));
 	flabel.set_color(color_);
 	flabel.set_position(xloc, yloc);
```

Comment:
Line 551's initialisation of `yloc` should have some scaling, otherwise the labels move north/south a bit as the zoom changes.